server/src/test/java/org/elasticsearch/index/IndexModuleTests.java

```diff
@@ -244,6 +246,39 @@ public void testRegisterIndexStore() throws IOException {
         indexService.close("simon says", false);
     }
 
+    public void testDirectoryWrapper() throws IOException {
```

Comment:
I see also a org.elasticsearch.index.IndexModuleTests#testFrozen that tests setReaderWrapper(). I think you can also test setDirectoryWrapper() there for frozen.

Reply:
Makes sense, I pushed f93acb4ff5c8e5db76f825a15b04b4c85c80b017